### ISSUE TITLE
Layout/ExtraSpacing with ForceEqualSignAlignment and endless methods with arguments and default values

### ISSUE THREAD

**Issue by dmke** [rubocop/rubocop#13751]
When using endless methods with default arguments, the Layout/ExtraSpacing cop wants to align the equal signs when `ForceEqualSignAlignment` is true. It seems it confuses the argument assignment and the endless method mark (not sure what to call the `=` in `def foo = 42`).

<details open><summary><code>mini_logger.rb</code></summary>

```rb
# frozen_string_literal: true

MiniLogger = Struct.new(:level) do
  def deleted(lvl = 1) = log(lvl) { "\e[31m[DEL]\e[0m #{yield}" }
  def updated(lvl = 1) = log(lvl) { "\e[36m[UPD]\e[0m #{yield}" }
  def ignored(lvl = 2) = log(lvl) { "\e[37m[IGN]\e[0m #{yield}" }
  def added(lvl = 1)   = log(lvl) { "\e[32m[ADD]\e[0m #{yield}" }

  def log(at)
    warn(yield) if at <= verbosity
  end
end
```

</details>
<details><summary><code>.rubocop.yml</code></summary>

```yaml
AllCops:
  NewCops: disable
  TargetRubyVersion: 3.2

Layout/ExtraSpacing:
  ForceEqualSignAlignment: true
```

</details>

## Expected behavior

RuboCop should not mark the equals sign of the method definition as "not aligned with the previous statement".

## Actual behavior

It does:

```console
$ bundle exec rubocop --safe --debug
For $PWD: configuration from $PWD/.rubocop.yml
Default configuration from $PWD/.gems/3.2/gems/rubocop-1.71.0/config/default.yml
Use parallel by default.
Running parallel inspection
Inspecting 2 files
Scanning $PWD/Gemfile
Loading cache from $HOME/.cache/rubocop_cache/070fa8d80082cded46bb8097b87d455d36025656/71ac05a58ecc6035d87b5359bb62b32bad9b98e9/06ca1808c44bd50645c0429eb11d7c4d6825f112
.Scanning $PWD/mini_logger.rb
Loading cache from $HOME/.cache/rubocop_cache/070fa8d80082cded46bb8097b87d455d36025656/71ac05a58ecc6035d87b5359bb62b32bad9b98e9/d97aed9f59671025f49108e201ea1b4ed9bc9dae
C

Offenses:

mini_logger.rb:5:24: C: Layout/ExtraSpacing: = is not aligned with the preceding assignment.
  def updated(lvl = 1) = log(lvl) { "\e[36m[UPD]\e[0m #{yield}" }
                       ^
mini_logger.rb:6:24: C: Layout/ExtraSpacing: = is not aligned with the preceding assignment.
  def ignored(lvl = 2) = log(lvl) { "\e[37m[IGN]\e[0m #{yield}" }
                       ^
mini_logger.rb:7:24: C: Layout/ExtraSpacing: = is not aligned with the preceding assignment.
  def added(lvl = 1)   = log(lvl) { "\e[32m[ADD]\e[0m #{yield}" }
                       ^

2 files inspected, 3 offenses detected
Finished in 0.25743580699963786 seconds
```

## Steps to reproduce the problem

- copy the `mini_logger.rb` and `.rubocop.yml` into a working directory
- run `bundle exec rubocop --safe`

## RuboCop version

```
$ bundle exec rubocop -V
1.71.0 (using Parser 3.3.7.0, rubocop-ast 1.37.0, analyzing as Ruby 3.2, running on ruby 3.2.6) [x86_64-linux]
```
